src/core/Akka.Persistence.Tests/JournalHealthCheckSpec.cs

```diff
@@ -0,0 +1,132 @@
+                                                       call-timeout = 1s
+                                                       reset-timeout = 10s
+                                                   }
+                                               }
+                                               akka.persistence.journal.failing-half-open {
+                                                   class = "Akka.Persistence.Tests.FailingJournal, Akka.Persistence.Tests"
+                                                   circuit-breaker {
+                                                       max-failures = 1
+                                                       call-timeout = 1s
+                                                       reset-timeout = 1s
+                                                   }
+                                               }
+                                               # Disable message serialization for circuit breaker tests to avoid serialization issues
+                                               akka.actor.serialize-messages = off
+
+                                   """;
+        return TestConfigs.TestSchedulerConfig
+            .WithFallback(Configuration("PersistenceHealthCheckSpec", extraConfig: extraConfig));
+    }
+
+    public JournalHealthCheckSpec(ITestOutputHelper output) : base(HealthCheckConfig(), output)
+    {
+    }
+
+    [Theory]
+    [InlineData(null)] // default plugin
+    [InlineData("akka.persistence.journal.inmem")]
+    public async Task JournalHealthCheck_should_default_to_Healthy(string? pluginId)
+    {
+        using var cts = new CancellationTokenSource(RemainingOrDefault);
+        var pluginHealth = await Extension.CheckJournalHealthAsync(pluginId, cts.Token);
+
+        Assert.Equal(PersistenceHealthStatus.Healthy, pluginHealth.Status);
+        Assert.NotNull(pluginHealth.Description);
+    }
+
+    [Fact]
+    public async Task JournalHealthCheck_should_return_Degraded_when_CircuitBreaker_is_Open()
+    {
+        // Get the journal actor reference
+        var journal = Extension.JournalFor("akka.persistence.journal.failing-open");
+
+        // Trigger a failure to open the circuit breaker
+        var writeMsg = new WriteMessages(new[] { new AtomicWrite(new Persistent("test", 1, "test-pid")) }.ToImmutableList(),
+            TestActor, 1);
+        journal.Tell(writeMsg, TestActor);
+
+        // Advance time to let the write fail and circuit breaker open
+        var testScheduler = (TestScheduler)Sys.Scheduler;
+        testScheduler.Advance(TimeSpan.FromSeconds(2));
+
+        using var cts = new CancellationTokenSource(RemainingOrDefault);
+        var pluginHealth = await Extension.CheckJournalHealthAsync("akka.persistence.journal.failing-open", cts.Token);
+
+        Assert.Equal(PersistenceHealthStatus.Degraded, pluginHealth.Status);
+        Assert.Contains("Circuit breaker is open", pluginHealth.Description);
```

Comment:
Probably not necessary to be this specific, but wanted to illustrate that leveraging the `CircuitBreaker` is a good default health heuristic.